snuba/web/views.py

```diff
@@ -447,15 +447,34 @@ def eventstream(*, dataset: Dataset):
         return ("ok", 200, {"Content-Type": "text/plain"})
 
     @application.route("/tests/<dataset:dataset>/drop", methods=["POST"])
-    def drop(*, dataset: Dataset):
+    def drop(*, dataset: Dataset) -> Tuple[str, int, Mapping[str, str]]:
         for storage in dataset.get_all_storages():
-            clickhouse = storage.get_cluster().get_query_connection(
-                ClickhouseClientSettings.MIGRATE
+            cluster = storage.get_cluster()
+            clickhouse = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
+            database = cluster.get_database()
+
+            tables_to_empty = set()
+
+            for schema in storage.get_schemas().get_unique_schemas():
+                if isinstance(schema, TableSchema):
+                    tables_to_empty.add(schema.get_local_table_name())
+
+            parts = clickhouse.execute(
+                """
+                SELECT partition, table
+                FROM system.parts
+                WHERE database = %(database)s
+                AND table IN (%(tables)s)
+                AND active = 1
+                """,
+                {"database": database, "tables": ", ".join(tables_to_empty)},
             )
-            for statement in storage.get_schemas().get_drop_statements():
-                clickhouse.execute(statement.statement)
 
-        ensure_tables_migrated(force=True)
+            for (part, table) in parts:
+                clickhouse.execute(
+                    f"ALTER TABLE {database}.{table} DROP PARTITION {part}"
+                )
```

Comment:
What was the rationale for this change? If this is not dropping the tables for the dataset anymore should we change the name?
Also if the goal is to empty the tables without dropping them why not just
`TRUNCATE TABLE sentry_local`
instead of deleting all the partitions ?

Reply:
Oh lol. I didn't know about the truncate statement, that's so much easier. I don't want to rename the endpoint since Sentry is referring to it as /drop. This change allows us to not need to figure out what was dropped and rerun the migrations on those tables everytime this endpoint gets hit.

Reply:
Updated to do truncate table instead. The old way won't work under the new system because the new migration system would just look at the state in the migrations table and do nothing (since all the migrations would be marked as completed there). It won't pick up that specific tables have been dropped when this endpoint gets hit.